packages/atomic/src/components/facets/facet-common.ts

```diff
@@ -23,3 +30,68 @@ export interface FacetValueProps {
   part?: string;
   buttonRef?: (element?: HTMLButtonElement) => void;
 }
+
+function isCategoryFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is CategoryFacetValueRequest {
+  return 'children' in value && Array.isArray(value.children);
+}
+
+function isFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is FacetValueRequest | CategoryFacetValueRequest {
+  return (
+    'value' in value &&
+    typeof value.value === 'string' &&
+    !('children' in value)
+  );
+}
+
+function getSelectedCategoryFacetValueRequest(
+  ancestor: CategoryFacetValueRequest
+): CategoryFacetValueRequest | null {
+  if (ancestor.state === 'selected') {
+    return ancestor;
+  }
+  for (const child of ancestor.children) {
+    const selectedValue = getSelectedCategoryFacetValueRequest(child);
+    if (selectedValue !== null) {
+      return selectedValue;
+    }
+  }
+  return null;
+}
+
+export function parseDependsOn(
+  dependsOn: Record<string, string>
+): AnyFacetDependency<AnyFacetValueRequest>[] {
+  return Object.entries(dependsOn).map(([parentFacetId, expectedValue]) => {
+    return {
+      parentFacetId,
+      isDependencyMet: (values) => {
+        return values.some((value) => {
+          if (isCategoryFacetValueRequest(value)) {
+            const selectedValue = getSelectedCategoryFacetValueRequest(value);
+            if (!selectedValue) {
+              return false;
+            }
+            if (!expectedValue) {
+              return true;
+            }
+            return selectedValue.value === expectedValue;
```

Comment:
This logic is repeated for both value types. You could just get the selectedValue/activeValue of either category (recursive) or specific facet (which would return null if idle).

Reply:
I could, but in this case, I don't think it would be easier to read, only less repetitive.